File: src/model/course-admin/CourseLessonResponse.ts
import { LessonType } from "@/types/enum/LessonType";
import { CodingExerciseDTO } from "./CodingExerciseDTO";
import { QuizDTO } from "./QuizDTO";

export interface CourseLessonResponse {
  lessonId: string;
  slug: string;
  title: string;
  description?: string;
  content?: string;
  orderIndex: number;
  lessonType: LessonType;
  isPreview: boolean;
  isPublished: boolean;
  videoUrl?: string;       // Embed URL (YouTube) hoặc PocketBase URL (cũ)
  youtubeVideoId?: string; // Video ID thuần, ví dụ: "DjlGte968ko"
  embedUrl?: string;       // Embed URL sẵn sàng nhúng iframe
  codingExercises: CodingExerciseDTO[];
  quizzes: QuizDTO[];

  totalPoints?: number;
}

File: src/main/java/com/truongsonkmhd/unetistudy/dto/course_dto/CourseLessonResponse.java
package com.truongsonkmhd.unetistudy.dto.course_dto;

import com.truongsonkmhd.unetistudy.common.LessonType;
import com.truongsonkmhd.unetistudy.dto.coding_exercise_dto.CodingExerciseDTO;
import com.truongsonkmhd.unetistudy.dto.quiz_dto.QuizDTO;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.util.List;
import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@FieldDefaults(level = AccessLevel.PRIVATE)
public class CourseLessonResponse {
    UUID lessonId;
    String title;
    Integer orderIndex;
    LessonType lessonType;
    Boolean isPreview;
    Boolean isPublished;
    String videoUrl;
    String description;
    String content;
    String slug;
    Integer totalPoints;
    Integer duration;

    List<CodingExerciseDTO> codingExercises;
    List<QuizDTO> quizzes;
}
